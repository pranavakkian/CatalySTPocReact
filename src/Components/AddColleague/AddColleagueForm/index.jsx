import { useState } from 'react';
import { Button, Dropdown, Form } from 'semantic-ui-react';

const AddColleagueForm = () => {
  const [username, setUsername] = useState('');
  const [contact, setContact] = useState('');
  const [jobtitle, setJobTitle] = useState('');
  const options = [
    { key: '01', text: 'Team Member', value: 'Team Member' },
    { key: '02', text: 'Developer', value: 'Developer' },

  ];
  return (
    <div>
      <Form>
        <Form.Field>
          <input id="username" placeholder="Username" label="Username" value={username} onChange={(e) => setUsername(e.target.value)} />
        </Form.Field>
        <Form.Field>
          <Dropdown
            fluid
            selection
            placeholder="Job title"
            data-testid="dropdown"
            options={options}
            value={jobtitle}
            onChange={(e, data) => setJobTitle(data.value)}
          />
        </Form.Field>
        <Form.Field>
          <input id="contact" placeholder="Contact" value={contact} label="Contact" onChange={(e) => setContact(e.target.value)} />
        </Form.Field>
        <Button type="submit">Submit</Button>
      </Form>
    </div>
  );
};

export default AddColleagueForm;
