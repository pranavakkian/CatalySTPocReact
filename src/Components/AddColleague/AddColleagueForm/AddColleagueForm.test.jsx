import {
  render,
  cleanup,
  screen,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AddColleagueForm from './index';

afterEach(cleanup);

const { default: userEvent } = require('@testing-library/user-event');

describe('AddColleagueForm', () => {
  test('Component rendered successfully', () => {
    const { container } = render(<AddColleagueForm />);
    expect(container).toMatchSnapshot();
  });
  test('handles username change correctly', () => {
    render(<AddColleagueForm />);
    expect(screen.getAllByRole('textbox')[0]).toHaveValue('');
    userEvent.type(screen.getAllByRole('textbox')[0], 'test');
    expect(screen.getAllByRole('textbox')[0]).toHaveValue('test');
  });

  test('handles contact change correctly', () => {
    render(<AddColleagueForm />);
    expect(screen.getAllByRole('textbox')[1]).toHaveValue('');
    userEvent.type(screen.getAllByRole('textbox')[1], '123');
    expect(screen.getAllByRole('textbox')[1]).toHaveValue('123');
  });

  test('handles Dropdown change correctly', () => {
    render(<AddColleagueForm />);
    userEvent.click(screen.getByRole('listbox'));
    userEvent.click(screen.getAllByRole('option')[0]);
    expect(screen.getByRole('listbox')).toHaveTextContent('Team Member');
    userEvent.click(screen.getByRole('listbox'));
    userEvent.click(screen.getAllByRole('option')[1]);
    expect(screen.getByRole('listbox')).toHaveTextContent('Developer');
  });
});
