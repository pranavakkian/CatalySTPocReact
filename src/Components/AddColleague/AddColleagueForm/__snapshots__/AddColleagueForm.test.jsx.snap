// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddColleagueForm Component rendered successfully 1`] = `
<div>
  <div>
    <form
      class="ui form"
    >
      <div
        class="field"
      >
        <input
          id="username"
          label="Username"
          placeholder="Username"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <div
          aria-expanded="false"
          class="ui fluid selection dropdown"
          data-testid="dropdown"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-atomic="true"
            aria-live="polite"
            class="divider default text"
            role="alert"
          >
            Job title
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <div
              aria-checked="false"
              aria-selected="true"
              class="selected item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Team Member
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                Developer
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <input
          id="contact"
          label="Contact"
          placeholder="Contact"
          value=""
        />
      </div>
      <button
        class="ui button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
